### ISSUE TITLE
DDev starts without error - browser says "Unable to connect"

### ISSUE THREAD

**Issue by JayFriendly** [ddev/ddev#5688]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
Jaypan:test jay$ ddev debug test
Running bash [-c /var/folders/kx/x5p46xn97yd70n32bp2z03c40000gs/T/test_ddev.sh]
======= Existing project config =========
These config files were loaded for project test: [/Users/jay/websites/test/.ddev/config.yaml]
name: test
type: drupal10
docroot: web
php_version: 8.1
webserver_type: nginx-fpm
webimage: ddev/ddev-webserver:v1.22.6
additional_hostnames: []
additional_fqdns: []
database: {mariadb 10.4}
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2
nodejs_version: 18
default_container_timeout: 120
======= Creating dummy project named  tryddevproject-24989 in ../tryddevproject-24989 =========
OS Information: Darwin Jaypan.local 23.2.0 Darwin Kernel Version 23.2.0: Wed Nov 15 21:54:55 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T8122 arm64
ProductName:		macOS
ProductVersion:		14.2.1
BuildVersion:		23C71
User information: uid=505(jay) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),79(_appserverusr),80(admin),81(_appserveradm),98(_lpadmin),333(piavpn),401(com.apple.access_screensharing-disabled),33(_appstore),100(_lpoperator),204(_developer),250(_analyticsusers),395(com.apple.access_ftp),399(com.apple.access_ssh),400(com.apple.access_remote_ae)
DDEV version:  ITEM             VALUE
 DDEV version     v1.22.6
 architecture     arm64
 db               ddev/ddev-dbserver-mariadb-10.4:v1.22.6
 ddev-ssh-agent   ddev/ddev-ssh-agent:v1.22.6
 docker           24.0.7
 docker-compose   v2.23.3
 docker-platform  colima
 mutagen          0.17.2
 os               darwin
 router           ddev/ddev-traefik-router:v1.22.6
 web              ddev/ddev-webserver:v1.22.6
PROXY settings: HTTP_PROXY='' HTTPS_PROXY='' http_proxy='' NO_PROXY=''
======= DDEV global info =========
Global configuration:
instrumentation-opt-in=true
omit-containers=[]
performance-mode=mutagen
router-bind-all-interfaces=true
internet-detection-timeout-ms=3000
disable-http2=false
use-letsencrypt=false
letsencrypt-email=
table-style=default
simple-formatting=false
use-hardened-images=false
fail-on-hook-fail=false
required-docker-compose-version=v2.23.3
use-docker-compose-from-path=false
project-tld=
xdebug-ide-location=
no-bind-mounts=false
router=traefik
wsl2-no-windows-hosts-mgt=false
router-http-port=80
router-https-port=443
mailpit-http-port=8025
mailpit-https-port=8026
traefik-monitor-port=10999

======= DOCKER info =========
docker location: lrwxrwxr-x  1 jay  admin  34 Jan  9 19:19 /opt/homebrew/bin/docker -> ../Cellar/docker/24.0.7/bin/docker
Docker Desktop Version: Print: Entry, ":CFBundleShortVersionString", Does Not Exist
Print: Entry, ":CFBundleVersion", Does Not Exist
Docker Desktop for Mac File build Doesn'tDocker Desktop for Mac Exist, build WillDocker Desktop for Mac Create: build /Applications/Docker.app/Contents/Info.plistDocker Desktop for Mac File build Doesn'tDocker Desktop for Mac Exist, build WillDocker Desktop for Mac Create: build /Applications/Docker.app/Contents/Info.plist
docker version:
Client: Docker Engine - Community
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.21.3
 Git commit:        afdd53b4e3
 Built:             Thu Oct 26 07:06:42 2023
 OS/Arch:           darwin/arm64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.7
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.10
  Git commit:       311b9ff
  Built:            Thu Oct 26 09:07:58 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.25
  GitCommit:        d8f198a4ed8892c764191ef7b3b06d8a2eeb5c7f
 runc:
  Version:          1.1.10
  GitCommit:        v1.1.10-0-g18a0cb0
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
DOCKER_DEFAULT_PLATFORM=notset
======= Mutagen Info =========
Mutagen is installed in ddev, version=0.17.2
--------------------------------------------------------------------------------
Name: drupal
Identifier: sync_k1GwFoqutkP7bcsqlLrU3UJp0UB4onRWcarVmcHuw4A
Labels:
	com.ddev.config-hash: c212281329fb24762a96e1b7a37a19ed84868866
	com.ddev.volume-signature: Users-jay-colima-default-1703396880
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/drupal
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-drupal-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: graphing
Identifier: sync_3zNLdGbFFmB7tujfQInagRIurp4zYZ282RgcPUa8cJ5
Labels:
	com.ddev.config-hash: c212281329fb24762a96e1b7a37a19ed84868866
	com.ddev.volume-signature: Users-jay-colima-default-1704566649
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/graphing
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-graphing-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: leveltest
Identifier: sync_yVJWXbHRXNeWASoq27BJgJBPc50zwvW9uvfmdSYL4Ta
Labels:
	com.ddev.config-hash: c212281329fb24762a96e1b7a37a19ed84868866
	com.ddev.volume-signature: Users-jay-colima-default-1704780334
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/leveltest
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-leveltest-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: test
Identifier: sync_jfQwp6Mg2k7dE7nGFgffN97OwHCfJoiNgF2djV4RHYx
Labels:
	com.ddev.config-hash: c212281329fb24762a96e1b7a37a19ed84868866
	com.ddev.volume-signature: Users-jay-colima-default-1704856539
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
		/web/sites/default/files
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/test
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-test-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: tryddevproject-30341
Identifier: sync_BKG9u8LyvVI3Bu1SLAGgs5rS9fzO8dpdF3g6Aq2L7fT
Labels:
	com.ddev.config-hash: 0311359b0f6eedb428c2679de60b5d021c2d717b
	com.ddev.volume-signature: Users-jay-colima-default-1704860184
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/tryddevproject-30341
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-tryddevproject-30341-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: tryddevproject-11551
Identifier: sync_sokuGDABMRhHqoUiOvdLroHcIfajf5im7ULQvMdJGLD
Labels:
	com.ddev.config-hash: 0311359b0f6eedb428c2679de60b5d021c2d717b
	com.ddev.volume-signature: Users-jay-colima-default-1704860281
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/tryddevproject-11551
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: No
Beta:
	URL: docker://ddev-tryddevproject-11551-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: No
Status: [Paused]
--------------------------------------------------------------------------------
Name: tryddevproject-27288
Identifier: sync_EGKnY7DfmGwUb2khGJHJHaUoRFgHWYYaOiVyiRiIilB
Labels:
	com.ddev.config-hash: 0311359b0f6eedb428c2679de60b5d021c2d717b
	com.ddev.volume-signature: Users-jay-colima-default-1704860408
Configuration:
	Synchronization mode: Two Way Resolved
	Hashing algorithm: Default (SHA-1)
	Maximum allowed entry count: Default (2⁶⁴−1)
	Maximum staging file size: Default (18 EB)
	Symbolic link mode: POSIX Raw
	Ignore VCS mode: Default (Propagate)
	Ignores:
		/.git
		/.tarballs
		/.ddev/db_snapshots
		/.ddev/.importdb*
		.DS_Store
		.idea
	Permissions mode: Default (Portable)
Alpha:
	URL: /Users/jay/websites/tryddevproject-27288
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
	Connected: Yes
	Synchronizable contents:
		28 directories
		78 files (107 kB)
		0 symbolic links
Beta:
	URL: docker://ddev-tryddevproject-27288-web/var/www/html
		DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
	Configuration:
		Watch mode: Default (Portable)
		Watch polling interval: Default (10 seconds)
		Probe mode: Default (Probe)
		Scan mode: Default (Accelerated)
		Stage mode: Neighboring
		File mode: Default (0600)
		Directory mode: Default (0700)
		Default file/directory owner: Default
		Default file/directory group: Default
		Compression: Default (DEFLATE)
	Connected: Yes
	Synchronizable contents:
		28 directories
		78 files (107 kB)
		0 symbolic links
Status: Watching for changes
--------------------------------------------------------------------------------
======= Docker Info =========
Docker platform: colima
Colima version: 0.6.7
Using Docker context: default (unix:///Users/jay/.colima/default/docker.sock)
docker-compose: v2.23.3
Using DOCKER_HOST=unix:///Users/jay/.colima/default/docker.sock
Docker version: 24.0.7
Able to run simple container that mounts a volume.
Docker disk space:
Filesystem                Size      Used Available Use% Mounted on
overlay                  95.8G     11.3G     84.5G  12% /

 Container ddev-tryddevproject-27288-db  Stopped
 Container ddev-tryddevproject-27288-web  Stopped
 Container ddev-tryddevproject-27288-db  Stopped
 Container ddev-tryddevproject-27288-web  Stopped
 Container ddev-tryddevproject-27288-db  Removed
 Container ddev-tryddevproject-27288-web  Removed
 Network ddev-tryddevproject-27288_default  Removed
Project tryddevproject-27288 has been stopped.
Stopped Mutagen daemon
The ddev-ssh-agent container has been removed. When you start it again you will have to use 'ddev auth ssh' to provide key authentication again.
Network ddev_default removed
Existing docker containers:
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
Network ddev_default created
Starting tryddevproject-24989...
Network ddev-tryddevproject-24989_default created
 Container ddev-ssh-agent  Created
 Container ddev-ssh-agent  Started
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys.
Building project images...
Project images built in 1s.
 Container ddev-tryddevproject-24989-web  Created
 Container ddev-tryddevproject-24989-db  Created
 Container ddev-tryddevproject-24989-web  Started
 Container ddev-tryddevproject-24989-db  Started
You have Mutagen enabled and your 'php' project type doesn't have `upload_dirs` set.
For faster startup and less disk usage, set upload_dirs to where your user-generated files are stored.
If this is intended you can disable this warning with `ddev config --disable-upload-dirs-warning`.
Starting Mutagen sync process...
Mutagen sync flush completed in 3s.
For details on sync status 'ddev mutagen st tryddevproject-24989 -l'
Waiting for web/db containers to become ready: [web db]
Starting ddev-router if necessary...
 Container ddev-router  Created
 Container ddev-router  Started
Waiting for additional project containers to become ready...
All project containers are now ready.
Successfully started tryddevproject-24989
Project can be reached at https://tryddevproject-24989.ddev.site https://127.0.0.1:32807
======== Curl of site from inside container:
HTTP/1.1 200 OK
Server: nginx
Date: Wed, 10 Jan 2024 04:23:57 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding

======== curl -I of http://tryddevproject-24989.ddev.site from outside:
curl: (7) Failed to connect to tryddevproject-24989.ddev.site port 80 after 1 ms: Couldn't connect to server
Unable to curl the requested project Please provide this output in a new gist at gist.github.com.
```
</details>         

### Expected Behavior

I expect to be able to access the site in a browser

### Actual Behavior

1. The browser says "Unable to connect"

### Steps To Reproduce

I have recently upgraded from an  Intel Mac to an Apple Silicon Mac. I migrated my user account, and now DDev is not working. I suspect something has not linked correctly.

Please note that I have uninstalled Homebrew Intell, reinstalled  Homebrew for Apple Silicon, reinstalled Docker, reinstalled Colima, and reinstalled DDev. Then I created a simple project with no modifications for testing. When I start the project, it looks fine:

```
Jaypan:test jay$ ddev start
Starting test...
Network ddev-test_default created
Building project images...
Project images built in 1s.
 Container ddev-test-web  Created
 Container ddev-test-db  Created
 Container ddev-test-web  Started
 Container ddev-test-db  Started
Starting Mutagen sync process...
...Mutagen sync flush completed in 7s.
For details on sync status 'ddev mutagen st test -l'
Waiting for web/db containers to become ready: [web db]
Starting ddev-router if necessary...
 Container ddev-router  Created
 Container ddev-router  Started
Waiting for additional project containers to become ready...
All project containers are now ready.
Successfully started test
Project can be reached at https://test.ddev.site https://127.0.0.1:32815
```

However, accessing the site in a browser fails, and cURL responds:
```
Jaypan:test jay$ curl http://test.ddev.site
curl: (7) Failed to connect to test.ddev.site port 80 after 4 ms: Couldn't connect to server
```

Can you please provide assistance?

### Anything else?

I have checked for something listening on ports 80 and 443, and I don't see any other services listening on those ports:

```
Jaypan:test jay$ sudo lsof -i -P | grep LISTEN | grep :80
Jaypan:test jay$ sudo lsof -i -P | grep LISTEN | grep :443
Jaypan:test jay$
```

Pinging the URL does return a response:
```
Jaypan:test jay$ ping test.ddev.site
PING kubernetes.docker.internal (127.0.0.1): 56 data bytes
64 bytes from 127.0.0.1: icmp_seq=0 ttl=64 time=0.153 ms
```

**Comment by rfay** [comment]
You've done a good job troubleshooting, and thanks for providing the output of `ddev debug test`, which helps enormously.

I see you're using Colima. I guess I would try doing a `colima delete` and then recreating the Colima instance per the docs.

Or consider using OrbStack, which is more predictable and reliable in general.

I do see something odd... on the host, if you `ping test.ddev.site` it wouldn't normally show it pinging `kubernetes.docker.internal`. I don't know where that reverse lookup comes from, you may want to remove any extra entries from your /etc/hosts file; that one definitely shouldn't be in there. 

And of course, the standard things that should be done any time you have networking trouble, temporarily turn off virus checker, firewall, VPN, proxies, and reboot.  This problem can absolutely be caused by firewall.

I think you've done these things, but it's worth looking again, https://ddev.readthedocs.io/en/latest/users/usage/troubleshooting/

**Comment by JayFriendly** [comment]
@alesbencina  - that you for the suggestion, that didn't work unfortunately.

@rfay  - that you for getting back to me. The problem was indeed a colima issue, and I needed to run `colima delete` to delete my Colima instance and rebuild it (Note to others - back up your database first if you are doing this). I also had a legacy reference to kubernetes in my hosts file that you pointed out, which wasn't causing the issue, but also wasn't needed and I removed it. 

So my problem is solved. Much appreciated.
